test/constructs/for_test.exs

```diff
@@ -55,15 +74,243 @@ defmodule Surface.Constructs.ForTest do
+          """
+        end
+
+      message =
+        ~S(code:3: invalid value for property "prop". Expected a :list, got: "some string".)
+
+      assert_raise(CompileError, message, fn ->
+        compile_surface(code)
+      end)
+    end
+  end
+
+  describe "#else language structure" do
+    test "renders inner `for` content if generator is not empty" do
+      html =
+        render_surface do
+          ~H"""
+          <#for each={fruit <- ["apples", "bananas", "oranges"]}>
+          <span>The inner content {fruit}</span>
+          <span>with multiple tags</span>
+          <#else>
+          <span>The else content</span>
+          <span>with multiple tags</span>
+          </#for>
+          """
+        end
+
+      assert html =~ """
+             <span>The inner content apples</span>
+             <span>with multiple tags</span>
+             <span>The inner content bananas</span>
+             <span>with multiple tags</span>
+             <span>The inner content oranges</span>
+             <span>with multiple tags</span>
+             """
+    end
+
+    test "renders inner content with component" do
+      html =
+        render_surface do
+          ~H"""
+          <#for each={fruit <- ["apples", "bananas", "oranges"]}>
+          <SomeComponent content={fruit} />
+          <#else>
+          <span>The else content</span>
+          <span>with multiple tags</span>
+          </#for>
+          """
+        end
+
+      assert html =~ """
+             <span>apples</span>
+             <span>bananas</span>
+             <span>oranges</span>
+             """
+    end
+
+    test "renders inner `for` content with assigns" do
+      fruits = ["apples", "bananas", "oranges"]
+      assigns = %{fruits: fruits}
+
+      html =
+        render_surface do
+          ~H"""
+          <#for each={fruit <- @fruits}>
+          <span>The inner content {fruit}</span>
+          <span>with multiple tags</span>
+          <#else>
+          <span>The else content</span>
+          <span>with multiple tags</span>
+          </#for>
+          """
+        end
+
+      assert html =~ """
+             <span>The inner content apples</span>
+             <span>with multiple tags</span>
+             <span>The inner content bananas</span>
+             <span>with multiple tags</span>
+             <span>The inner content oranges</span>
+             <span>with multiple tags</span>
+             """
+    end
+
+    test "raises compile error with complex generator" do
+      assigns = %{list1: [1, 4], list2: [2, 3, 4], range: 1..3}
+
+      code =
+        quote do
+          ~H"""
+          <#for each={x <- @list1, y <- @list2, x in @range, y in @range}>
+          <span>x: {x}, y: {y}</span>
+          <#else>
+          <span>The else content</span>
+          <span>with multiple tags</span>
+          </#for>
+          """
+        end
+
+      message = ~r/using <#for>..<#else> only supports single generators without filters./
```

Comment:
We should include the `file:line` part in the regex to make sure the error is raised at the expected line.